features/device-config/persistence/api/src/main/java/org/opennms/features/deviceconfig/persistence/api/DeviceConfigDao.java

```diff
@@ -36,8 +36,10 @@
 
 public interface DeviceConfigDao extends OnmsDao<DeviceConfig, Long> {
 
-    List<DeviceConfig> findConfigsForInterfaceSortedByDate(OnmsIpInterface ipInterface);
+    List<DeviceConfig> findConfigsForInterfaceSortedByDate(OnmsIpInterface ipInterface, ConfigType configType);
 
-    Optional<DeviceConfig> getLatestConfigForInterface(OnmsIpInterface ipInterface);
+    Optional<DeviceConfig> getLatestSucceededConfigForInterface(OnmsIpInterface ipInterface, ConfigType configType);
```

Comment:
I am not sure, if we need that method that considers only for succeeded device configs.